### ISSUE TITLE
Trusted Network Detection

### ISSUE THREAD

**Issue by PeterMcCaldon-aps** [Timac/VPNStatus#21]
the VPNutil gives us the option to ignore SSID's but it would be nice if it could also ignore specific network ranges or ideally if it detects an internal resource (ping or https connection)

**Comment by Timac** [comment]
Interesting. I assume that you never want to connect to the VPN if a ping to a resource success or fail?
Do you have a more precise example?

**Comment by PeterMcCaldon-aps** [comment]
Yes, I'm working on an Always on VPN project which seems very difficult on a Mac. Love the VPNStatus app btw.

I'm trying to get it so the user doesn't need to do anything. So if they are at home the VPN connects but if they are in the office it even doesn't try to connect

If the app could check for a resource before trying to connect that should tell it if they are in the office (not needing the VPN) or outside (needing the VPN)

The resource could be checking for an internal website, so does https://myweb.mydomain.local succeed or can I ping the DC

It's kinda like the ignore SSID in the VPNUtil, if 10.99.99.* is detected don't connect. The above example is how many other AoVPN providers do it.

Thanks

**Comment by Timac** [comment]
@PeterMcCaldon-aps Thanks for the clarification. Such a feature could be implemented in VPNStatus. As a first idea, I would most likely change `-(BOOL)shouldPreventAutoConnectOnCurrentSSID` to return the BOOL asynchronously in a block. The block would then asynchronously execute `[neService connect];` if the resource is available.

The function `shouldPreventAutoConnectOnCurrentSSID` could be expanded to not only check for the SSIDs to be ignored but also test if the resource is available. Finally a `preventConnectIfResourceDetected` function containing the list of resources to check could be added similar to `ignoredSSIDs`:

```
-(NSArray<NSString *> *)ignoredSSIDs
{
    NSString *ignoredSSIDsString = [[NSUserDefaults standardUserDefaults] stringForKey:kServiceIgnoredSSIDsKey];
    return [ignoredSSIDsString componentsSeparatedByString:@","];
}
```
